docs/release-process.md

```diff
@@ -11,109 +11,89 @@
+
+   - First determine whether there are any "non-consensus-breaking changes that require a fresh chainstate".
+      - In other words, the database schema has changed, but an automatic migration was not implemented.
+      - Determine whether this a feature release, as opposed to a hotfix or a patch.
+   - A new branch in the format `release/X.Y.Z.A.n` is created from the base branch `develop`.
+
+2. Enumerate PRs and/or issues that would _block_ the release.
+
+   - A label should be applied to each such issue/PR as `X.Y.Z.A.n-blocker`.
+
+3. Since development is continuing in the `develop` branch, it may be necessary to cherry-pick some commits into the release branch.
+
+   - Create a feature branch from `release/X.Y.Z.A.n`, ex: `feat/X.Y.Z.A.n-pr_number`.
+   - Add cherry-picked commits to the `feat/X.Y.Z.A.n-pr_number` branch
+   - Merge `feat/X.Y.Z.A.n-pr_number` into `release/X.Y.Z.A.n`.
+
+4. Open a PR to update the [CHANGELOG](../CHANGELOG.md) file in the `release/X.Y.Z.A.n` branch.
+
+   - Create a chore branch from `release/X.Y.Z.A.n`, ex: `chore/X.Y.Z.A.n-changelog`.
+   - Add summaries of all Pull Requests to the `Added`, `Changed` and `Fixed` sections.
+      - Pull requests merged into `develop` can be found [here](https://github.com/stacks-network/stacks-core/pulls?q=is%3Apr+is%3Aclosed+base%3Adevelop+sort%3Aupdated-desc).
+        
+        **Note**: GitHub does not allow sorting by _merge time_, so, when sorting by some proxy criterion, some care should be used to understand which PR's were _merged_ after the last release.
+
+5. Once `chore/X.Y.Z.A.n-changelog` has merged, a build may be started by manually triggering the [`CI` workflow](../.github/workflows/ci.yml) against the `release/X.Y.Z.A.n` branch.
+
+6. Once the release candidate has been built and binaries are available, ecosystem participants shall be notified to test the tagged release on various staging infrastructure.
+
+7. The release candidate will tested that it successfully syncs with the current chain from genesis both in testnet and mainnet.
+
+8. If bugs or issues emerge from the rollout on staging infrastructure, the release will be delayed until those regressions are resolved.
+
+   - As regressions are resolved, additional release candidates should be tagged.
+   - Repeat steps 3-7 as necessary.
+
+9. Once the final release candidate has rolled out successfully without issue on staging infrastructure, the tagged release shall no longer marked as Pre-Release on the [Github releases](https://github.com/stacks-network/stacks-core/releases/) page.
+   Announcements will then be shared in the `#stacks-core-devs` channel in the Stacks Discord, as well as the [mailing list](https://groups.google.com/a/stacks.org/g/announce).
+
+10. Finally, the following merges will happen to complete the release process:
+    - Release branch `release/X.Y.Z.A.n` will be merged into the `master` branch.
+    - Then, `master` will be merged into `develop`.
```

Comment:
Maybe here we could add a tip to "watch" this repository releases so users also get notifications on GH?

Reply:
I disagree - that may be something we could add to the README. this document is more about what shall be done to ship a release.

Reply:
That makes sense, I'll resolve this.